src/shims/unix/linux/eventfd.rs

```diff
@@ -74,28 +78,33 @@ impl FileDescription for Event {
 
         // Block when counter == 0.
         let counter = self.counter.get();
+        let weak_eventfd = self_ref.downgrade();
         if counter == 0 {
             if self.is_nonblock {
                 return ecx.set_last_error_and_return(ErrorKind::WouldBlock, dest);
             }
+            let dest = dest.clone();
+            let mut blocked_read_tid = self.blocked_read_tid.borrow_mut();
+            *blocked_read_tid = Some(ecx.active_thread());
```

Comment:
Yea I missed that and that's indeed a possible case. Multithreaded case is a bit more complicated, I need more time to think about this, but this is the case I am considering: 

If there are main thread + 2 additional threads: 
let the counter be u64::max, so any ``write`` happened after will block. 
2. thread1 try to write, and get blocked.
3. thread2 try to write, also get blocked.
4. Main thread read. Now, will thread1 and thread2 both get unblocked? 

If both are unblocked, if thread1 write a u64::max again, thread2 should be blocked again. But the current implementation won't allow this to happen because thread2 is currently calling the unblock callback.

It also doesn't make sense to only unblock one thread, because the moment ``read`` happens, both thread1 and thread2 indeed can write.

I will check what actually happened in real system.

Reply:
I think eventfd will unblock every thread at once. In this case, if this happens 

If there are main thread + 2 additional threads:
1. let the counter be (u64::max - 1), so any write happened after will block.
2. thread1 try to write, and get blocked.
3. thread2 try to write, also get blocked.
4. main thread 1, all thread gets unblocked
5. thread1 unblocked and writes (u64::max - 1)
6. thread 2 unblocked and try to continue writing 1, it gets block again

So essentially, we need to check the value and continue to block the thread again in the blocking callback functions. I am not sure how we can approach this without overcomplicated it because if we need to block a thread multiple times, we might run into something like

1. block_thread
2. in the blocking callback, block_thread
3. In the second block_thread callback, block_thread again
... and it continues

Reply:
You can have a `fn eventfd_write` that blocks the thread if necessary, and in the "unblock" callback calls `eventfd_write` again. Recursion is a thing. :)

Reply:
Yea, I think this will work. I feel this has an unfortunate consequence of making it more complex, but I couldn't see another solution for this.